### DISCUSSION TITLE
In the PulseUI -> Network -> Request are always Pending

### DISCUSSION THREAD

**Discussion by mackoj** [kean/Pulse#112] [Q&A]
Hi 👋,

When I try to monitor my network request they all seem to be always **Pending**. I would be glad to know if someone has an idea to fix this.

Thanks,

![Simulator Screen Shot - iPhone 12 - 2022-09-16 at 16 47 16](https://user-images.githubusercontent.com/661647/190667182-04b2294a-6006-4364-ac9b-badba2e81e47.png)

**Comment by mackoj** [comment]
I use Pulse 2.1.2 configured using a **sharedURLSession** because the `URLSessionProxyDelegate.enableAutomaticRegistration()` seems to work.

```
    let sharedURLSession = URLSession(configuration: .default, delegate: URLSessionProxyDelegate(), delegateQueue: nil)
```

This is how I initialise my globalLogger.
```
  let bundleName = Bundle.main.object(forInfoDictionaryKey: "CFBundleIdentifier") as! String
  LoggingSystem.bootstrap(PersistentLogHandler.init)
  URLSessionProxyDelegate.enableAutomaticRegistration()
  return Logger(label: bundleName)
```

My network client is implemented with [swift-url-routing](https://github.com/pointfreeco/swift-url-routing) it use 👇 to make request.
`URLSession.data(for request: URLRequest, delegate: URLSessionTaskDelegate? = nil) async throws -> (Data, URLResponse)`

Do you have some idea @kean to make this work ?

Thanks,

**Reply by kean** [reply]
Hi, are you using `URLSession.shared`? If yes, it's not supported.

**Reply by kean** [reply]
I suggest debugging `NetworkLogger` to see if this code executes:

```swift
public func logTask(_ task: URLSessionTask, didCompleteWithError error: Error?) {
    guard error != nil || !configuration.isWaitingForDecoding else { return }
    _logTask(task, didCompleteWithError: error)
}
```

**Reply by kean** [reply]
Potentially related https://github.com/kean/Pulse/issues/113

**Reply by mackoj** [reply]
> Hi, are you using `URLSession.shared`? If yes, it's not supported.

No, I use `let sharedURLSession = URLSession(configuration: .default, delegate: URLSessionProxyDelegate(), delegateQueue: nil)`.

> I suggest debugging NetworkLogger to see if this code executes:

It don't seem to be executed.

> Potentially related https://github.com/kean/Pulse/issues/113

Yes I do think it's related thanks I will follow the issue.
